journal.py

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import os
+from pyramid.config import Configurator
+from pyramid.view import view_config
+from waitress import serve
+import sqlalchemy as sa
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import scoped_session, sessionmaker
+from zope.sqlalchemy import ZopeTransactionExtension
+import datetime
+from pyramid.httpexceptions import HTTPFound
+from sqlalchemy.exc import DBAPIError
+from pyramid.authentication import AuthTktAuthenticationPolicy
+from pyramid.authorization import ACLAuthorizationPolicy
+from pyramid.security import remember, forget
+from cryptacular.bcrypt import BCRYPTPasswordManager
+
+DBSession = scoped_session(sessionmaker(extension=ZopeTransactionExtension()))
+
+
+DATABASE_URL = os.environ.get(
+    'DATABASE_URL',
+    'postgresql://jason@localhost:5432/learning-journal')
+
+HERE = os.path.dirname(os.path.abspath(__file__))
+
+Base = declarative_base()
+
+
+@view_config(route_name='home', renderer='templates/index.jinja2')
+def list_view(request):
+    entries = Entry.all()
+    return {'entries': entries}
+
+
+@view_config(route_name='detail', renderer='templates/detail.jinja2')
+def detail_view(request):
+    article_id = request.matchdict['id']
+    article = Entry.get_article(article_id)
+    title = article[0].title
+    body_text = article[0].body_text
+    created = article[0].created
+    return {'title': title, 'body_text': body_text, 'created': created,
+            'article_id': article_id}
```

Comment:
In which case I would access them as object attributes in the template? That sounds much better; hadn't realized I could do that.